app/code/Magento/InventorySales/Plugin/InventoryReservationsApi/PreventAppendReservationOnNotManageItemsInStockPlugin.php

```diff
@@ -61,7 +61,7 @@ public function aroundExecute(AppendReservationsInterface $subject, \Closure $pr
                 $reservation->getStockId()
             );
 
-            if ($stockItemConfiguration === null || $stockItemConfiguration->isManageStock()) {
+            if ($stockItemConfiguration->isManageStock()) {
```

Comment:
this is not correct fix, because of condition `$stockItemConfiguration === null` was added to support Order processing for deleted products from Catalog.

and condition `$stockItemConfiguration === null` checks that product does not exists anymore 

`isProductAssignedToStock::execute($sku, $stockId)` should be used here as well, but Exception should be caught here and Reservation should be appended anyway